### DISCUSSION TITLE
artillery-engine-playwright

### DISCUSSION THREAD

**Discussion by ChetanKB** [artilleryio/artillery#1437] [Artillery]
Hello guys,
I am executing the playwright JS script via artillery engine and for single async function  its working but How can we execute multiple Tests via artillery this is my project structure and 
final.spec.js contains all tests

![project structure](https://user-images.githubusercontent.com/53107584/167132168-592f285f-9f79-4db0-9f57-d5566949c7c0.PNG)

**Comment by hassy** [comment]
hi @ChetanKB 

you can specify more than one Artillery scenario in your YAML file, each of which can load a separate Playwright script.

More information on the `scenarios` section here: https://www.artillery.io/docs/guides/guides/test-script-reference#scenarios-section

What does your Artillery YAML look like at the moment?

**Reply by ChetanKB** [reply]
Hi @hassay  this what my yaml looks like and after executing playwright tests also we need to post logs to datadog
config:
  target: https://www.wikipedia.org
  plugins:
    expect:
      outputFormat: "json"

    metrics-by-endpoint:
      useOnlyRequestNames: true

    publish-metrics:
      - port: "{{ $processEnvironment.DD_STATSD_PORT }}"

        prefix: "platformqa.playwright."

        host: "{{ $processEnvironment.DD_AGENT_HOST }}"

        type: "datadog"

        event:
          title: "Platform QA Automated Test"

          priority: "low"

          send: true

          status: "Info"

          tags:
            - "test_suite:{{ $processEnvironment.TEST_SUITE }}"

            - "bu:{{ $processEnvironment.BU }}"

            - "squad:{{ $processEnvironment.SQUAD }}"

            - "target_service:{{ $processEnvironment.TARGET_SERVICE }}"

            - "source:platformqa.apitestrunner"

  engines:
    playwright: {}

  processor: "./wiki-latest.js"

scenarios:
  - engine: playwright

    flowFunction: "wikipedia"

 # - think: 1

  #  flowFunction: "amazon"

    flow: []

**Reply by hassy** [reply]
That should work as long as your `wiki-latest.js` exports the two functions referenced by scenarios:

1. `wikipedia`
2. `amazon`

Those can be defined inside `wiki-latest.js` or imported from other modules (e.g. modules under `test/`). As long as they're exported from `wiki-latest.js` it will work.
